### PR TITLE
Add .editorconfig

### PR DESCRIPTION
**Pull request by williamboman**
http://editorconfig.org/

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+[*]
+indent_style = tab
+
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (JocelynDelalande) [inline .editorconfig]: why?

Comment (williamboman) [inline .editorconfig]: https://github.com/caskroom/homebrew-cask/pull/13662#discussion_r38876366

Comment (JocelynDelalande) [inline .editorconfig]: thanks, didn't know :) 
(that's a silly-designed feature of markdown btw :-/)

**Comment by JocelynDelalande** [comment]
:+1: for me

**Comment by astorije** [comment]
I have never used this myself, I'd prefer @erming or @floogulinc to step in.

**Comment by floogulinc** [comment]
Looks OK to me

**Comment by astorije** [comment]
Thanks @williamboman! (Catching up on Shout a bit :-) )

**Pull request merged**